### PR TITLE
#160617673 add articles  app to authors project for creating articles

### PR DESCRIPTION
**Pull request by AmosWels**
#### What does this PR do?
-  Enables a user to create their articles, edit, read/access articles created by other authors.
-  Forbids a user from updating and deleting articles that they haven't authored
#### Description of Task to be completed?
- When a user has been authenticated successfully, they can create articles, update their own articles and view other peoples articles /api/articles/  authors.apps.articles.views.ArticleAPIView
#### How should this be manually tested?
- Creating articles by accessing this endpoint at ```/api/articles/``` in postman
-  Edit, delete articles with this endpoint at ```/api/articles/<str:slug>/ ```in postman
#### Screenshots
- create articles
image

- delete article
image

- view single article
image

- view all articles
image

- update article
image

#### What are the relevant pivotal tracker stories?
- #160617673

### CODE REVIEW

**Review by PatrickCmd — COMMENTED** [review]
@AmosWels Register the articles app in admin site

File: authors/apps/articles/views.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+from rest_framework import status
+from rest_framework.permissions import IsAuthenticated,IsAuthenticatedOrReadOnly
+from rest_framework.response import Response
+from authors.apps.articles.renderers import ArticleJSONRenderer
+from authors.apps.articles.serializers import ArticleSerializer
+from rest_framework.exceptions import PermissionDenied
+from .models import Article
+from rest_framework import serializers
+from django.core.exceptions import ObjectDoesNotExist
+# Create your views here.
```
Comment (PatrickCmd) [inline authors/apps/articles/views.py]: By the PEP8 guide lines, it's always a good practice to have two white space lines left before the definition of class.

File: authors/apps/articles/views.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        article = request.data.get("article", {})
+        serializer = self.get_serializer(data=article)
+        serializer.is_valid(raise_exception=True)
+        self.perform_create(serializer)
+        headers = self.get_success_headers(serializer.data)
+        return Response(serializer.data, status=status.HTTP_201_CREATED, headers=headers)
+    
+    def perform_create(self, serializer):
+        serializer.save(author=self.request.user)
+class ArticleAPIDetailsView(generics.RetrieveUpdateDestroyAPIView):
```
Comment (PatrickCmd) [inline authors/apps/articles/views.py:33]: Same as above

File: authors/apps/articles/views.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+    renderer_classes = (ArticleJSONRenderer,)
+    serializer_class = ArticleSerializer
+    lookup_field = "slug"
+    queryset = Article.objects.all()
+    
+    def destroy(self, request, *args, **kwargs):
+        instance = self.get_object()
+        if instance.author != request.user:
+            raise PermissionDenied
```
Comment (PatrickCmd) [inline authors/apps/articles/views.py:44]: Pass message parameter in the **permissionDenied** exception.

File: authors/apps/articles/views.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        self.perform_destroy(instance)
+        return Response({"message":"Comment Deleted"},status=status.HTTP_204_NO_CONTENT)
+
+    def update(self, request, *args, **kwargs):
+        article_dict = request.data.get("article", {})
+        partial = kwargs.pop('partial', False)
+        instance = self.get_object()
+        if instance.author != request.user:
+            raise PermissionDenied
```
Comment (PatrickCmd) [inline authors/apps/articles/views.py:53]: Same as above.

File: authors/apps/articles/views.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    serializer_class = ArticleSerializer
+    lookup_field = "slug"
+    queryset = Article.objects.all()
+    
+    def destroy(self, request, *args, **kwargs):
+        instance = self.get_object()
+        if instance.author != request.user:
+            raise PermissionDenied
+        self.perform_destroy(instance)
+        return Response({"message":"Comment Deleted"},status=status.HTTP_204_NO_CONTENT)
```
Comment (kcharles52) [inline authors/apps/articles/views.py]: Thanks for the good work, check the message we are supposed to return article deleted

**Review by PatrickCmd — APPROVED** [review]
LGTM

**Pull request merged**